extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantConfigBean.java

```diff
@@ -1,66 +1,115 @@
 
+import org.jboss.logging.Logger;
+
 import io.quarkus.arc.BeanDestroyer;
+import io.quarkus.oidc.OIDCException;
 import io.quarkus.oidc.OidcTenantConfig;
 import io.smallrye.mutiny.Uni;
 
 public class TenantConfigBean {
+    private static final Logger LOG = Logger.getLogger(TenantConfigBean.class);
 
     private final Map<String, TenantConfigContext> staticTenantsConfig;
     private final Map<String, TenantConfigContext> dynamicTenantsConfig;
     private final TenantConfigContext defaultTenant;
-    private final Function<OidcTenantConfig, Uni<TenantConfigContext>> tenantConfigContextFactory;
+    private final TenantConfigContextFactory tenantConfigContextFactory;
+
+    @FunctionalInterface
+    public interface TenantConfigContextFactory {
+        Uni<TenantConfigContext> create(OidcTenantConfig tenantConfig, boolean dynamicTenant);
+    }
 
     public TenantConfigBean(
             Map<String, TenantConfigContext> staticTenantsConfig,
             Map<String, TenantConfigContext> dynamicTenantsConfig,
             TenantConfigContext defaultTenant,
-            Function<OidcTenantConfig, Uni<TenantConfigContext>> tenantConfigContextFactory) {
+            TenantConfigContextFactory tenantConfigContextFactory) {
         this.staticTenantsConfig = staticTenantsConfig;
         this.dynamicTenantsConfig = dynamicTenantsConfig;
         this.defaultTenant = defaultTenant;
         this.tenantConfigContextFactory = tenantConfigContextFactory;
     }
 
-    public Map<String, TenantConfigContext> getStaticTenantsConfig() {
+    public TenantConfigContext getDefaultTenant() {
+        return defaultTenant;
+    }
+
+    public Map<String, TenantConfigContext> getStaticTenants() {
         return staticTenantsConfig;
     }
 
-    public TenantConfigContext getDefaultTenant() {
-        return defaultTenant;
+    public Map<String, TenantConfigContext> copyStaticTenantsWithDefault() {
+        HashMap<String, TenantConfigContext> staticWithDefault = new HashMap<>(staticTenantsConfig);
+        staticWithDefault.put(OidcUtils.DEFAULT_TENANT_ID, defaultTenant);
+        return staticWithDefault;
+    }
+
+    public TenantConfigContext getStaticTenantConfig(String tenantId) {
+        return staticTenantsConfig.get(tenantId);
     }
 
-    public Function<OidcTenantConfig, Uni<TenantConfigContext>> getTenantConfigContextFactory() {
-        return tenantConfigContextFactory;
+    public boolean isStaticTenant(String tenantId) {
+        return staticTenantsConfig.containsKey(tenantId);
     }
 
-    public Map<String, TenantConfigContext> getDynamicTenantsConfig() {
-        return dynamicTenantsConfig;
+    public Uni<TenantConfigContext> getOrCreateTenantContext(OidcTenantConfig oidcConfig, boolean dynamicTenant) {
+        // check if the connection has already been created
+        Map<String, TenantConfigContext> tenantsConfig = dynamicTenant ? dynamicTenantsConfig : staticTenantsConfig;
```

Comment:
For dynamic tenants, the connection is established during the request, so AFAIK, their context always must be in the ready state if it is in recorded in the map